app/scripts/modules/core/src/managed/ArtifactsList.tsx

```diff
@@ -134,14 +136,29 @@ function getArtifactStatuses({ environments }: IManagedArtifactVersion): Artifac
   // NOTE: The order in which entries are added to `statuses` is important. The highest priority
   // item must be inserted first.
```

Comment:
the way i'm stacking these right now is:

1. pending constraints — this feels like a 'things are actively happening and you should know about it' case
2. failed constraints — it feels kind of yucky to potentially not see these if other things are pending/running, but I thought 1 was more important
3. pinned state — important, but hopefully if you don't get a chance to see it here you'll see it in the overview or via our snazzy new warning card on other versions. I'm less worried about people missing this info if there are multiple status bubbles